app/models.py

```diff
@@ -497,6 +498,38 @@ def generate_oauth_client_id(client_name) -> str:
     return generate_oauth_client_id(client_name)
 
 
+class MfaBrowser(db.Model, ModelMixin):
+    user_id = db.Column(db.ForeignKey(User.id, ondelete="cascade"), nullable=False)
+    token = db.Column(db.String(64), default=False, nullable=False)
+    expires = db.Column(ArrowType, default=False, nullable=False)
+
+    user = db.relationship(User)
+
+    @classmethod
+    def create_new(cls, user, token_length=64) -> "MfaBrowser":
+        return MfaBrowser.create(
+            user_id=user.id,
+            token=random_string(token_length),
```

Comment:
if the `token` is unique, we might need to do a loop until a unique token is found, something like https://github.com/simple-login/app/blob/ec5821d89bb4bcf9ea70966745b6731cdabb3012/app/models.py#L1174